src/transformers/tokenization_utils.py

```diff
@@ -1079,6 +1079,10 @@ def _decode(
         self._decode_use_source_tokenizer = kwargs.pop("use_source_tokenizer", False)
 
         filtered_tokens = self.convert_ids_to_tokens(token_ids, skip_special_tokens=skip_special_tokens)
+        # If given is a single id, return the decoded text without further processing
+        if isinstance(filtered_tokens, str):
+            return filtered_tokens
+
```

Comment:
Nice!! Only thing we are missing is that if the single token is special, it should get skipped on:

```python 
.decode(special_token, skip_special_tokens=True)
```
We can update to:

```python
if isinstance(filtered_tokens, str):
    filtered_tokens = [filtered_tokens] # prevents splitting the string in following loop
```

+ we can test this in the test you added `test_decoding_extra_params`

Reply:
I played a bit with skipping special tokens and looks like the branch works as expected. The prev line includes a skipper :grin:  , looks like it makes the work already. 

```
filtered_tokens = self.convert_ids_to_tokens(token_ids, skip_special_tokens=skip_special_tokens)
```

I added more tests for this issue!

Reply:
I added a better example of the case below, let me know if you agree! `skip_special_tokens=True` should skip special tokens like `[PAD]` which it does when the input is a list, but not when the input is a single id. Let me know what your thoughts are! 😄 

```python
# 0 is token_id of '[PAD]'

BertTokenizer.from_pretrained("google-bert/bert-base-cased").decode(0, skip_special_tokens=True)
# This branch: '[PAD]'
# Expected: ''
```

Reply:
Got it now, made the changes and added the test! I think the branch is ready for merge @itazap

Reply:
agreed! @DuyguA Sorry I just added a few minor cosmetic suggestions to the names of the tests ! 😅  then we merge 👍

Reply:
No worries :grin:  One tiny thing is that, CI fails for some reason by only changing the test files :see_no_evil:

Reply:
ah okay it's an unrelated test that was failing on main, but is fixed now! Needs a rebase and should be good!😅

Reply:
> ah okay it's an unrelated test that was failing on main, but is fixed now! Needs a rebase and should be good!😅

Ah OK, rebased now, hope merged before new commits came :grin: :grin:

Reply:
@DuyguA the same (unrelated) tests are still failing here although have been fixed on main, can you please try to pull and merge / rebase again? 🙏

Reply:
Rebased @itazap  , all tests are passing now :wink: